cmd/influx/write.go

```diff
@@ -74,41 +90,102 @@ func cmdWrite(f *globalFlags, opt genericCLIOpts) *cobra.Command {
 	return cmd
 }
 
+func (writeFlags *writeFlagsType) dump(args []string) {
+	if writeFlags.Debug {
+		writeFlags.args = args
+		log.Printf("WriteFlags%+v", *writeFlags)
+	}
+}
+
 // createLineReader uses writeFlags and cli arguments to create a reader that produces line protocol
-func (writeFlags *writeFlagsType) createLineReader(args []string) (r io.Reader, closer io.Closer, err error) {
-	if len(args) > 0 && args[0][0] == '@' {
+func (writeFlags *writeFlagsType) createLineReader(cmd *cobra.Command, args []string) (io.Reader, io.Closer, error) {
+	readers := make([]io.Reader, 0, 2*len(writeFlags.Headers)+2*len(writeFlags.Files)+1)
+	closers := make([]io.Closer, 0, len(writeFlags.Files))
```

Comment:
closers are subset of readers herein, only opened files will be closed, nothing more ... stdin or string readers cannot/must not be closed.

Reply:
I hear you, seems better to do it as ReadClosers with a noop closer on those that don't need a closer, typical pattern in golang.

```go
type noopCloser struct {
    r io.reader
}

func (n *noopCloser) Close() error {
    return nil 
}
```

my big concern here is it makes it easy to lose track of closers/readers when they ares disparate uncoupled return types.

Last thing, please don't resolve conversations, I'll happily resolve them when I feel the conversation has come to fruition 👍

Reply:
Thank you for the clarification around "conversation resolving", as a newbie herein, I need such advice 👍

I see two concerns herein:
1. Why are readers and closers separate instead of being ReadClosers?
   - files can be provided in non-UTF-8 encoding, a file (ReaderCloser) is then transformed to UTF-8 Reader (not a ReadCloser) 
https://github.com/influxdata/influxdb/blob/b786fc16520aae61482f7e09c149b850aa27ff06/cmd/influx/write.go#L141
   - the last file or stdin can be then changed so that first N lines are skipped when --skipHeader option is present. This transformation also works with Readers, but not ReadClosers https://github.com/influxdata/influxdb/blob/b786fc16520aae61482f7e09c149b850aa27ff06/cmd/influx/write.go#L168   
   - changing the above intermediate  Readers to ReadClosers to have just one ReadCloser array would introduce an extra complexity herein IMO 
2. Why are io.Reader and io.Closer disparate return values?
   - I understand that a composite io.ReaderCloser looks better. It would require to wrap multiple Readers and Closers (or ReadClosers) in a new type that will only delegate calls. I wanted to prefer builtins whenever possible, but there is no multi-reader-closer for reuse.
  - At the same time, it is not a big deal (in my POV) for a non-public function to avoid this wrapping. "A returned Closer must be closed" is an expected contract IMO.